### PR TITLE
Added Email to header

### PR DESCRIPTION
**Pull request by yzainee**

### CODE REVIEW

**Comment by msrb** [comment]
Any reason for making `pom.xml` and `Bayesian.java` files executable? Permissions changed from 644 to 755. IDE playing tricks here maybe?

File: pom.xml [hunk 1]
```diff
@@ -78,6 +78,11 @@
       <artifactId>dnsjava</artifactId>
       <version>2.1.8</version>
     </dependency>
+    <dependency>
+	  <groupId>commons-logging</groupId>
```
Comment (msrb) [inline pom.xml]: I don't see this dependency used in the code - do we need it?

File: src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java [hunk 2]
```diff
@@ -46,6 +50,8 @@
 /* package */ class Bayesian {
 
     private static final String DEFAULT_BAYESIAN_URL = "https://recommender.api.openshift.io/";
+    private static final String OSIO_USERS_URL = "https://api.openshift.io/api/users";
```
Comment (msrb) [inline src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java]: This is good enough for testing purposes, but the URL should be configurable as the value is specific to prod deployment.

File: src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java [hunk 3]
```diff
@@ -140,6 +147,68 @@ public String getApiUrl() {
...
+        
+        Gson gson;
+        User responseObj;
+
+        HttpGet httpGet = new HttpGet(url);
+        try (CloseableHttpClient client = HttpClients.createDefault();
+                CloseableHttpResponse response = client.execute(httpGet)) {
+
+	            HttpEntity entity = response.getEntity();
+	          	InputStream is = entity.getContent();
```
Comment (msrb) [inline src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java]: I believe you're mixing tabs and spaces in this method. Please stick with spaces :)

File: src/main/java/com/redhat/jenkins/plugins/bayesian/User.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package com.redhat.jenkins.plugins.bayesian;
+
+
+import java.util.List;
+
+
+public class User {
```
Comment (msrb) [inline src/main/java/com/redhat/jenkins/plugins/bayesian/User.java]: This class is full of tabs :) Please switch to spaces to stay consistent with the rest of the codebase.

File: src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java (hunk 3 above)
Comment (msrb) [reply src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java]: I think this InputStream should be closed once we are done with it.

File: pom.xml (hunk 1 above)
Comment (yzainee) [reply pom.xml]: HttpGet was using this internally and build was failing without this dependency. Hence i added it

File: src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java [hunk 5]
```diff
@@ -140,6 +147,83 @@ public String getApiUrl() {
...
+        Gson gson;
+        User responseObj;
+        InputStream is = null;
+        BufferedReader br = null;
+
+        HttpGet httpGet = new HttpGet(url);
+        try (CloseableHttpClient client = HttpClients.createDefault();
+                CloseableHttpResponse response = client.execute(httpGet)) {
+    
+        	    HttpEntity entity = response.getEntity();
```
Comment (msrb) [inline src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java]: One more tab here :)

File: src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java [hunk 6]
```diff
@@ -140,6 +147,83 @@ public String getApiUrl() {
...
+                String line;
+                br = new BufferedReader(new InputStreamReader(is, "UTF-8"));
+
+                while ((line = br.readLine()) != null) {
+                    sb.append(line);
+                }
+
+                gson = new GsonBuilder().create();
+                 
+                responseObj = gson.fromJson(sb.toString(), User.class);
```
Comment (msrb) [inline src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java]: We do something similar in this file already: https://github.com/yzainee/fabric8-analytics-jenkins-plugin/blob/d5af23f1f2baa49cebdddbd5c1653388fe53e8bc/src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java#L119-L125

I believe you could reuse the same approach here. `try-with-resources` would also take care of closing the input stream for you.

**Pull request merged**